src/controllers/product.controller.js

```diff
@@ -230,4 +223,29 @@ const updateOnadd = async (req, res) => {
   }
 };
 
-export default { CreateCollection, DeleteCollection, addproduct, updateOnadd };
+const searchProducts = asyncWrapper(async (req, res) => {
+  const products = await productsServices.searchproduct(
+    req.query.key,
+    req.query.minPrice,
+    req.query.maxPrice
+  );
+  if (products.length === 0) {
+    return res.status(404).json({
+      code: 404,
+      message: 'product not found',
+    });
+  }
```

Comment:
we can remove this check, it not much important as we are searching. We can return empty array